### PR TITLE
Frontend typing

### PR DESCRIPTION
**Pull request by morganbelous**
### Summary

Got rid of most of the 'any' types for the frontend.

- [x] fixed frontend typing

### Notes

- APIWrapper.ts and Search.tsx still have any types, not sure what to do about those.

### CODE REVIEW

File: backend/src/api.ts [hunk 1]
```diff
@@ -105,41 +105,32 @@ router.post('/logout', (req: Request, res: Response) => {
...
   const handled = await getMember(req, res);
   res.status(handled!.status).json(handled);
 });
-router.post('/setMember', async (req: Request, res: Response) => {
+router.post('/setMember', async (req, res) => {
   const handled = await setMember(req, res);
   res.status(handled!.status).json(handled);
 });
-router.delete('/deleteMember/:email', async (req: Request, res: Response) => {
+router.delete('/deleteMember', async (req, res) => {
```
Comment (SamChou19815) [inline backend/src/api.ts:120]: mistake in merge?

File: backend/src/api.ts [hunk 2]
```diff
@@ -105,41 +105,32 @@ router.post('/logout', (req: Request, res: Response) => {
...
-  const handled = await setTeam(req, res);
-  res.status(handled!.status).json(handled);
-});
-router.post('/deleteTeam', async (req: Request, res: Response) => {
-  const handled = await deleteTeam(req, res);
-  res.status(handled!.status).json(handled);
-});
+router.get('/allTeams', allTeams);
+router.post('/setTeam', setTeam);
+router.delete('/deleteTeam', deleteTeam);
```
Comment (SamChou19815) [inline backend/src/api.ts:133]: another mistake in merge?

File: backend/src/memberAPI.ts [hunk 3]
```diff
@@ -161,14 +161,13 @@ export const deleteMember = async (
...
-    const { email } = req.params;
-    if (!email || email === '') {
+    if (!req.body.email || req.body.email === '') {
       return {
         status: 400,
         error: "Couldn't delete member with undefined email!"
       };
     }
-    const result = await MembersDao.deleteMember(email);
+    const result = await MembersDao.deleteMember(req.body.email);
```
Comment (SamChou19815) [inline backend/src/memberAPI.ts:170]: mistake?

File: frontend/src/API/APIWrapper.ts [hunk 4]
```diff
@@ -33,21 +33,6 @@ export default class APIWrapper {
...
-  ): Promise<any> {
-    const responseProm = axios
-      .delete(url, {
-        ...config,
-        withCredentials: true
-      })
-      .catch((err) => err)
-      .then((resOrErr) => this.responseMiddleware(resOrErr, errDefault));
-    return responseProm;
-  }
```
Comment (SamChou19815) [inline frontend/src/API/APIWrapper.ts:49]: why deleting this?

File: backend/src/api.ts (hunk 1 above)
Comment (morganbelous) [reply backend/src/api.ts:120]: I think I pulled the master branch changes into this branch once Henry fixed the delete member and edit teams problems? I didn't make these changes not sure what happened

Comment (SamChou19815) [reply backend/src/api.ts:120]: I think it's better to remove all the changes in the backend in this PR, since they are almost all unintended.

**Review by henry-li-06 — CHANGES_REQUESTED** [review]
Seems like as @SamChou19815 mentioned, some of the changes reverse the fixes from #37. The backend changes and the APIWrapper changes should probably be removed.

File: backend/src/api.ts (hunk 1 above)
Comment (morganbelous) [reply backend/src/api.ts]: just went back and fixed all the changes, it should be good now

**Review by SamChou19815 — APPROVED** [review]

**Review by henry-li-06 — APPROVED** [review]

**Pull request merged**